File: bin/src/main/java/be/zlz/kara/bin/services/webhook/WebhookAuthentication.java
package be.zlz.kara.bin.services.webhook;

import be.zlz.kara.bin.domain.enums.AuthenticationType;

public class WebhookAuthentication {

    private AuthenticationType type;

    private String identifier;

    private String secret;

    public WebhookAuthentication(AuthenticationType type, String identifier, String secret) {
        this.type = type;
        this.identifier = identifier;
        this.secret = secret;
    }

    public AuthenticationType getType() {
        return type;
    }

    public String getIdentifier() {
        return identifier;
    }

    public String getSecret() {
        return secret;
    }
}

File: bin/src/main/java/be/zlz/kara/bin/modules/webhook/WebhookAuthentication.kt
package be.zlz.kara.bin.modules.webhook

import be.zlz.kara.bin.domain.enums.AuthenticationType
import be.zlz.kara.bin.dto.v11.OAuth2GrantType

data class WebhookAuthentication(
        val type: AuthenticationType,
        val username: String?,
        val password: String?,
        val token: String?,
        val clientId: String?,
        val secret: String?,
        val grantType: OAuth2GrantType?
)